### PR TITLE
Reference implementations for ReduceLogicalAnd & ReduceLogicalOr

### PR DESCRIPTION
**Pull request by tomdol**

### CODE REVIEW

File: ngraph/src/ngraph/op/reduce_logical_and.cpp [hunk 1]
```diff
@@ -34,3 +36,39 @@ shared_ptr<Node> op::v1::ReduceLogicalAnd::clone_with_new_inputs(const OutputVec
...
+    void evaluate_reduce_logical_and(const HostTensorPtr& data,
+                                     const HostTensorPtr& axes,
+                                     const HostTensorPtr& out)
+    {
+        const auto axes_count = axes->get_element_count();
+        const auto axes_buffer = axes->get_data_ptr<int64_t>();
+        const AxisSet reduction_axes(
+            std::vector<AxisSet::value_type>(axes_buffer, axes_buffer + axes_count));
+
+        runtime::reference::reduce_logical_and(data->get_data_ptr<char>(),
```
Comment (ilyachur) [inline ngraph/src/ngraph/op/reduce_logical_and.cpp:51]: Are data types always equal to char for logical and?
Can the input data be int64 or something like that?

**Comment by gladilov-gleb** [comment]
@tomdol @Maxim-Doronin 

Could we enable tests in myriad plugin that were disabled due to lack of `evaluate` function to check implementation on plugin and nGraph are OK?

File: ngraph/src/ngraph/op/reduce_logical_and.cpp (hunk 1 above)
Comment (tomdol) [reply ngraph/src/ngraph/op/reduce_logical_and.cpp:51]: Both `ReduceLogicalAnd` and `ReduceLogicalOr` inherit from the `LogicalReduction` class. Its validation method makes sure that the input data type is `boolean` https://github.com/openvinotoolkit/openvino/blob/master/ngraph/src/ngraph/op/util/logical_reduction.cpp#L113-L115

The boolean element type is 8-bits long and is usually represented by `char` https://github.com/openvinotoolkit/openvino/blob/master/ngraph/src/ngraph/type/element_type.cpp#L78 and in fact this is how it was already done in constant folding passes for those ops prior to my changes.

**Comment by Maxim-Doronin** [comment]
@gladilov-gleb, @ilyachur, @tomdol, enabling related tests in myriad plugin is not trivial for now due requirement of tests refactoring and it makes no sense to do this in this PR. I'd merge it as is. 
I'm going to enable them as part of #-34278

**Pull request merged**